xclim/subset.py

```diff
@@ -164,6 +165,37 @@ def func_checker(*args, **kwargs):
     return func_checker
 
 
+def check_latlon_dimnames(func):
+    @wraps(func)
+    def func_checker(*args, **kwargs):
+        """
+        A decorator examining the names of the latitude and longitude dimensions and renames them temporarily if needed.
+        """
+
+        if range(len(args)) == 0:
+            return func(*args, **kwargs)
+
+        formatted_args = list()
+        for argument in args:
+            try:
+                if isinstance(argument, (xarray.DataArray, xarray.Dataset)):
+                    dims = argument.dims
+                else:
+                    raise TypeError
+            except (IndexError, TypeError):
```

Comment:
I might be missing something, but what could raise a `IndexError` here? Are there cases of DataArrays or Datasets without `dims`?

Reply:
For that, I can't say for sure. I imagine that an empty Dataset is possible. I figured that's a reasonable case that could happen.

Reply:
I suggest explicit type checking and error handling is done only in cases where they provide added-value: 
- the standard error message is unclear
- there is a potential for silent errors

Reply:
Type-checking here does serve a purpose: In the case of shape_subset, a shape can currently be passed as an arg. This prevents us from trying to grab dims from an unopened vector file so that much of the code is needed. The IndexError can definitely be removed.